librna/rnalib.c

```diff
@@ -54,6 +55,8 @@
 #include "ViennaRNA/params/io.h"
 #include "ViennaRNA/fold_vars.h"
 
+#define LOOKUP_SIZE 10000
```

Comment:
Why 10k? Is there a reason for that number?

Reply:
The biggest possible number that can be passed to the `scale` function during runtime is the length of the input sequence. Unfortunately passing the length of the input sequence to `scale` would've been too complicated, so we assume that input sequences should be shorter than 10000 bases most of the time, which is why we chose to precalculate the first 10000 scale values. If the input sequence is longer than 10000 bases, the scale values >= 10000 will be calculated on the spot.